src/db/sysdb_sudo.c

```diff
@@ -852,6 +852,56 @@ sysdb_sudo_add_sss_attrs(struct sysdb_attrs *rule,
     return EOK;
 }
 
+static errno_t sysdb_sudo_add_lowered_users(struct sss_domain_info *domain,
+                                            struct sysdb_attrs *rule)
+{
+    TALLOC_CTX *tmp_ctx;
+    const char **users = NULL;
+    const char *lowered = NULL;
+    errno_t ret;
+
+    if (domain->case_sensitive == true || rule == NULL) {
+        return EOK;
+    }
+
+    tmp_ctx = talloc_new(NULL);
+    if (tmp_ctx == NULL) {
+        return ENOMEM;
+    }
+
+    ret = sysdb_attrs_get_string_array(rule, SYSDB_SUDO_CACHE_AT_USER, tmp_ctx,
+                                       &users);
+    if (ret != EOK) {
+        DEBUG(SSSDBG_OP_FAILURE, "Unable to get %s attribute [%d]: %s\n",
+                SYSDB_SUDO_CACHE_AT_USER, ret, strerror(ret));
+        goto done;
+    }
+    if (users == NULL) {
+        return EOK;
+    }
+
+    for (int i = 0; users[i] != NULL; i++) {
+        lowered = sss_tc_utf8_str_tolower(tmp_ctx, users[i]);
+        if (lowered == NULL || strcmp(users[i], lowered) == 0) {
+            continue;
```

Comment:
What does `lowered == NULL` mean? I didn't find anything useful in libunistring documentation.
If NULL means an error, would be nice to split the if, print a debug message and treat the error accordingly.
 
By the way, it's not related to your code, but I personally don't like comparing the lowered string with users[i] to check whether the users[i] string was already a lower cased string or not. Would be better to do this check before converting the string, I know there's a u8_islowercase() function in libunistring and I hope it's being used somewhere down in the stack. I didn't find anything similar in GLib though.